### ISSUE TITLE
seeques - There may be a situation where atomic auction seller might not get his prefunded amount back due to blockstuffing or traffic congestion

### ISSUE THREAD

**Issue by sherlock-admin4** [sherlock-audit/2024-03-axis-finance-judging#139]
seeques

medium

# There may be a situation where atomic auction seller might not get his prefunded amount back due to blockstuffing or traffic congestion

## Summary
It is possible for atomic auction type to be prefunded. If the seller wishes to `cancel()` his lot, the remaining prefunded amount will be sent to him. The call to `cancel()` should happen before the lot conclusion. However, there exists a possibility where the tx to `cancel()` gets blockstuffed or is just sent during high chain traffic or sequencer downtime and submitted after the auction's conclusion timestamp. In that case, the remaining prefunding would be lost to the seller as there is no other options to get it back
## Vulnerability Detail
In the `cancelAuction()` function there is a modifier that restricts calling it when the `timpestamp > conclusion`:
```solidity
 function cancelAuction(uint96 lotId_) external override onlyInternal {
        // Validation
        _revertIfLotInvalid(lotId_);
        _revertIfLotConcluded(lotId_); //@audit

        // Call internal closeAuction function to update any other required parameters
        _cancelAuction(lotId_);

        // Update lot
        Lot storage lot = lotData[lotId_];
        lot.conclusion = uint48(block.timestamp);
        lot.capacity = 0;
    }
```
```solidity
    function _revertIfLotConcluded(uint96 lotId_) internal view virtual {
        // Beyond the conclusion time
        if (lotData[lotId_].conclusion < uint48(block.timestamp)) {
            revert Auction_MarketNotActive(lotId_);
        }
  
        // Capacity is sold-out, or cancelled
        if (lotData[lotId_].capacity == 0) revert Auction_MarketNotActive(lotId_);
    }
```
So, in case when small capacity isn't sold and there is a little time before conclusion, the seller might decide to cancel the auction and receive the remaining prefunded amount. However, due to the above described blockstuffing or traffic congestion or sequencer downtime, the transaction will be submitted after the conclusion timestamp and would revert, resulting in a loss of funds for the seller.
## Impact
Loss of funds
## Code Snippet
https://github.com/sherlock-audit/2024-03-axis-finance/blob/main/moonraker/src/modules/Auction.sol#L351-L364
https://github.com/sherlock-audit/2024-03-axis-finance/blob/main/moonraker/src/modules/Auction.sol#L733-L741
## Tool used

Manual Review

## Recommendation
For atomic auction type the cancel functionality should also work after the conclusion timestamp. One way to do that is to check the auction type in `cancelAuction()` function and allow calls to cancel after the conclusion timestamp.

Duplicate of #94
